ext/exif/exif.c

```diff
@@ -4756,17 +4755,24 @@ PHP_FUNCTION(exif_thumbnail)
 	exif_error_docref(NULL EXIFERR_CC, &ImageInfo, E_NOTICE, "Returning thumbnail(%d)", ImageInfo.Thumbnail.size);
 #endif
 
-	ZVAL_STRINGL(return_value, ImageInfo.Thumbnail.data, ImageInfo.Thumbnail.size);
-	if (arg_c >= 3) {
-		if (!ImageInfo.Thumbnail.width || !ImageInfo.Thumbnail.height) {
+	RETVAL_STRINGL(ImageInfo.Thumbnail.data, ImageInfo.Thumbnail.size);
+	if (z_width) {
+		if (!ImageInfo.Thumbnail.width) {
 			if (!exif_scan_thumbnail(&ImageInfo)) {
 				ImageInfo.Thumbnail.width = ImageInfo.Thumbnail.height = 0;
 			}
 		}
 		ZEND_TRY_ASSIGN_REF_LONG(z_width,  ImageInfo.Thumbnail.width);
+	}
+	if (z_height) {
+		if (!ImageInfo.Thumbnail.height) {
+			if (!exif_scan_thumbnail(&ImageInfo)) {
```

Comment:
Now you'll be scanning twice in case of a zero height.
I also don't like the duplication.
I'd just do (untested):
```c
if ((z_width || z_height) && (!ImageInfo.Thumbnail.width || !ImageInfo.Thumbnail.height)) {
			if (!exif_scan_thumbnail(&ImageInfo)) {
				ImageInfo.Thumbnail.width = ImageInfo.Thumbnail.height = 0;
			}
}
```